### ISSUE TITLE
CognitoUser.deleteUser(GenericHandler handler) doesn't call handler for success.

### ISSUE THREAD

**Issue by AngusMorton** [aws/aws-sdk-android#304]
In the Cognito Identity Provider, the `deleteUser` method does not call the provided `GenericHandler` when it has succeeded.

```java
public void deleteUser(GenericHandler callback) {
  if (callback == null) {
    throw new CognitoParameterInvalidException("callback is null");
  }
  
  try {
    deleteUserInternal(this.getCachedSession());
  } catch (final Exception e) {
    callback.onFailure(e);
  }
}
```

Should probably be:

```java
public void deleteUser(GenericHandler callback) {
  if (callback == null) {
    throw new CognitoParameterInvalidException("callback is null");
  }
  
  try {
    deleteUserInternal(this.getCachedSession());
    callback.onSuccess();
  } catch (final Exception e) {
    callback.onFailure(e);
  }
}
```

This would make the API consistent with `deleteUserInBackground`.

**Comment by kvasukib** [comment]
Hi Angus,

Thank you for pointing this out. We will take a look at it.

**Comment by kvasukib** [comment]
Fixed in 2.6.5 release.
